Improved the SVG graphic and code.

denilsonsa opened the pull request.

denilsonsa committed:
Rewriting the SVG code from scratch.

denilsonsa committed:
Updating README and removing PNG images.

* Gittip is now Gratipay
* badgr.co is now shields.io
* We don't need the PNG images.

Maybe I should have removed the gittip donation link altogether, but I
wouldn't do it with permission from the author.

denilsonsa committed:
Using third-party https://rawgit.com/ to deliver the SVG file with the correct headers.

https://stackoverflow.com/questions/13808020/include-an-svg-hosted-on-github-in-markdown

twolfson commented on coderwall.svg:
```diff
@@ -1,58 +1,10 @@
+<circle cx="270" cy="30" r="30" fill="#3e8dcc"/>
+<circle cx="150" cy="150" r="30" fill="#3e8dcc"/>
+<circle cx="270" cy="150" r="30" fill="#3e8dcc"/>
+<circle cx="270" cy="270" r="30" fill="#3e8dcc"/>
```
This doesn't seem to be rendering in GitHub's SVG preview. Can you find out what's wrong with that?

denilsonsa replied:
[Works fine for me now](https://github.com/denilsonsa/coderwall-svg/blob/master/coderwall.svg). Earlier today GitHub had some server issues.

twolfson replied:
It's strange. It wasn't loading in Firefox on OS X but is loading in Firefox on Linux fine. Weird.

twolfson commented on coderwall.svg:
```diff
@@ -1,58 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Drawn from scratch, based on coderwall.png from https://assembly.com/coderwall/assets -->
+<svg version="1.1" viewBox="0 0 300 300" xmlns="http://www.w3.org/2000/svg">
+<circle cx="30" cy="30" r="30" fill="#3e8dcc"/>
```
Please indent these `circle` tags. I prefer XML tags to have indentation for each level of depth.

twolfson commented on README.md:
```diff
@@ -1,10 +1,5 @@
-Coderwall SVG [![Donate on Gittip](http://badgr.co/gittip/twolfson.png)](https://www.gittip.com/twolfson/)
+Coderwall SVG [![Donate on Gittip](https://img.shields.io/gratipay/twolfson.svg)](https://gratipay.com/~twolfson/)
```
This pattern is very out of date with how I typically link to gratipay. Please leave this in its original form and I will update it accordingly in another PR/release

twolfson commented on README.md:
```diff
@@ -1,10 +1,5 @@
-Download Links
---------------
-
-[svg](https://raw.github.com/twolfson/coderwall-svg/master/coderwall.svg) [png](https://raw.github.com/twolfson/coderwall-svg/master/coderwall.png)
```
Please move back to the original split format of logo and download links. It was much easier for someone to get a link to download the resource (e.g. don't need to click to the page then click "raw" then click download).

twolfson commented on README.md:
```diff
@@ -1,10 +1,5 @@
-
-[svg](https://raw.github.com/twolfson/coderwall-svg/master/coderwall.svg) [png](https://raw.github.com/twolfson/coderwall-svg/master/coderwall.png)
\ No newline at end of file
+[![Coderwall logo](https://cdn.rawgit.com/denilsonsa/coderwall-svg/88921c6211e60c960cd8c154068e1688affaa97b/coderwall.svg)](coderwall.svg)
```
Please move back to `raw.github.com`, this was intentional as:
- There is no caching when viewing the resource meaning `master` is always up to date
- We don't need to worry about bumping the SHA on every edit to the SVG

twolfson commented:
This PR was very aggressive and prob could have avoided a lot of work if an issue was opened first. The majority of requests are to move everything back to their original format and only update the SVG.

denilsonsa committed:
Fixing comments from pull request.

Reverted most changes to README.md.

Added back a PNG snapshot just because GitHub can't render SVG images.

denilsonsa commented:
> This PR was very aggressive and prob could have avoided a lot of work if an issue was opened first. The majority of requests are to move everything back to their original format and only update the SVG.

Don't worry. Reverting the `README.md` file is easy. And the SVG file has been updated.

twolfson commented on coderwall.svg:
```diff
@@ -1,58 +1,9 @@
+	<circle cx="270" cy="30" r="30" fill="#3e8dcc"/>
+	<circle cx="150" cy="150" r="30" fill="#3e8dcc"/>
+	<circle cx="270" cy="150" r="30" fill="#3e8dcc"/>
+	<circle cx="270" cy="270" r="30" fill="#3e8dcc"/>
```
Please use spaces for indentation as that was the existing coding style for this repo

denilsonsa committed:
Tabs to 2 spaces.

denilsonsa commented:
Done.

twolfson commented:
Sorry to keep on making this a back and forth. It looks like there is no whitespace padding as part of the SVG but in most scenarios, the logo has padding. For example:

https://assembly.firesize.com/120x120/g_center/https://d1015h9unskp4y.cloudfront.net/attachments/1e4aebe6-4771-45da-8805-693787bfe644/coderwall.png

Is there anything in the Coderwall style guide about how much padding is required?

denilsonsa committed:
Using sizes and padding from:

* http://codepen.io/TimPietrusky/pen/hLolB
* https://coderwall.com/p/z5wuia/coderwall-logo-made-with-html-css

denilsonsa commented:
There is no actual official style guide. I've rewritten the dimensions to the closest-to-official I could get.

denilsonsa committed:
Updating the PNG file.

twolfson commented on coderwall.svg:
```diff
@@ -1,58 +1,9 @@
-     style="fill:#000000;fill-opacity:1;stroke:#000000;stroke-opacity:1" />
+<?xml version="1.0" encoding="utf-8"?>
+<svg version="1.1" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
+  <circle cx="5" cy="5" r="1.4142" fill="#3e8dcc"/>
```
For our reference, what fraction is `1.4142`?

denilsonsa replied:
Analyzing the CSS from http://codepen.io/TimPietrusky/pen/hLolB, the entire bg has `2em` of width, and each dot is `1em`, with `0.5em` offset from each one. Multiplying all by 10, we have the `viewBox="0 0 20 20"` and each center on `5` SVG points increments.

The radius was a bit more tricky. The CSS used `20%`; but what does that percentage mean? It is measured from the center to the furthest corner, which is the diagonal of the square. `20% × 0.5 × √2 ≅ 0.14142135623`

I'm pretty sure that `20%` is just some approximation by the author of that CSS example, but all "official" logos are low-quality raster pictures anyway.

twolfson replied:
Cool, thanks for the info =)

twolfson commented:
Looks great, thanks for working on this =)
